plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestClient.java

```diff
@@ -70,23 +89,20 @@ public void setHosts(HttpHost... hosts)
     public Response performRequest(String method, String endpoint, Header... headers)
             throws IOException
     {
-        try {
-            return Failsafe.with(retryPolicy).get(() -> delegate.performRequest(method, endpoint, headers));
-        }
-        catch (FailsafeException e) {
-            Throwable throwable = e.getCause();
-            if (throwable instanceof IOException) {
-                throw (IOException) throwable;
-            }
-            throw (RuntimeException) throwable;
-        }
+        return unwrapException(() -> delegate.performRequest(method, endpoint, headers));
     }
 
     public Response performRequest(String method, String endpoint, Map<String, String> params, HttpEntity entity, Header... headers)
             throws IOException
+    {
+        return unwrapException(() -> delegate.performRequest(method, endpoint, params, entity, headers));
+    }
+
+    private Response unwrapException(CheckedSupplier<Response> supplier)
```

Comment:
this belongs to previous commit

Reply:
I wanted to do a squash before merge. The only reason I split this in two commits is that it makes multiple review rounds easier, since you know you've reviewed a commit already.